### PR TITLE
geo-rep: Add util script to remove geo-rep location and corresponding image locations (PROJQUAY-4995)

### PR DESCRIPTION
**Pull request by jonathankingfc**
- This removes first the ImageStoragePlacements that are associated with the location
- Once those are cleaned up, the ImageStorageLocation itself is removed from the database
- To run this command use `python -m util.removelocation <location_name>`

### CODE REVIEW

**Comment by HammerMeetNail** [comment]
@jonathankingfc Can you add some notes on how this should be executed?

**Comment by jonathankingfc** [comment]
@HammerMeetNail added instruction to description

**Comment by jonathankingfc** [comment]
> This operation could potentially be dangerous and could lead to data loss. The script should log what was deleted (at least to stdout) and it should ask for confirmation before running. The user should absolutely **CONFIRM** during execution that they want this storage engine removed.

Added a y/n confirmation to the script

File: util/removelocation.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        else:
+            sys.stdout.write("Please respond with 'yes' or 'no' " + "(or 'y' or 'n').\n")
+
+def remove_location():
+    # Get the location name from the command line
+    location_name = sys.argv[1]
+    
+    if not query_yes_no(f"WARNING: This is a destructive operation. Are you sure you want to remove {location_name} from your storage locations?"):
+        print("Canceling operation...")
+        return
```
Comment (kleesc) [inline util/removelocation.py:29]: You should be using argparse for these types of command line options.

File: util/removelocation.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    
+    # Get all image placements for the location
+    query = (
+        ImageStoragePlacement
+        .where(ImageStoragePlacement.location == location_name)
```
Comment (kleesc) [inline util/removelocation.py:34]: Shouldn't it be `ImageStoragePlacement.location == location_id`?

File: util/removelocation.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    )
+
+    # Delete all image placements for the location
+    for manifest in query:
+        manifest.delete().where(ImageStoragePlacement.id == manifest.id).execute()
+
+    # Delte storage location from database
```
Comment (kleesc) [inline util/removelocation.py:41]: Typo

File: util/removelocation.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    
+    # Get all image placements for the location
+    query = (
+        ImageStoragePlacement
+        .where(ImageStoragePlacement.location == location_name)
+    )
+
+    # Delete all image placements for the location
+    for manifest in query:
+        manifest.delete().where(ImageStoragePlacement.id == manifest.id).execute()
```
Comment (kleesc) [inline util/removelocation.py:39]: These aren't manifests. Did you mean placement?

File: util/removelocation.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    # Delte storage location from database
+    ImageStorageLocation.delete().where(ImageStorageLocation.name == location_name).execute()
```
Comment (kleesc) [inline util/removelocation.py:42]: Probably want to log this somehow.

**Review by kleesc — DISMISSED** [review]

**Review by Sunandadadi — APPROVED** [review]
lgtm

**Pull request merged**